### DISCUSSION TITLE
Migration guide from 3.4.0 to 4.1.0

### DISCUSSION THREAD

**Discussion by maheshkurmi** [dyn4j/dyn4j#167] [General]
Lot has changed from version 3 to version 4 !
Is there any migration guide/complete change-log from version 3.4.0 to 4.1.0 ?
As my project has physics engine derived from dyn4j and is synchronised upto 3.4.0 and it has scripting API to access world, body, joints and collision data. There seem some awesome improvements in version 4, which worth included in my project.

**Comment by wnbittle** [comment]
I have not created a guide, but it should be straight forward - just link to the new version of the library and then fix any deprecation warnings by switching to the new classes/methods.  It mostly involves switching from classes in the `org.dyn4j.dynamics` package to classes in the `org.dyn4j.world package`.

You can see an example of this by reviewing the changes to the samples project https://github.com/dyn4j/dyn4j-samples/commit/0074fac86b30bb5a50bb1b8ac6951a7548b0d0fc.
